lib/view/commissievoorkeuren/CommissieVoorkeurenProfiel.php

```diff
@@ -28,20 +37,26 @@ public function getTitel() {
 
 	public function view() {
 		echo '<h1>' . $this->getTitel() . ' </h1>';
-		echo '<p>Naam: ' . ProfielModel::getLink($this->id, 'volledig') . '</p>';
-		$voorkeur = new CommissieVoorkeurenModel($this->id);
-		$voorkeuren = $voorkeur->getVoorkeur();
-		$commissies = $voorkeur->getCommissies();
+		echo '<p>Naam: ' . $this->profiel->getLink('volledig'). '</p>';
+
+		$opmerking = VoorkeurOpmerkingModel::instance()->getOpmerkingVoorLid($this->profiel);
+		$voorkeuren = CommissieVoorkeurModel::instance()->getVoorkeurenVoorLid($this->profiel);
+		$voorkeurenMap = array();
+		foreach($voorkeuren as $voorkeur) {
+		    $voorkeurenMap[$voorkeur->cid] = $voorkeur;
+        }
+		$commissies = VoorkeurCommissieModel::instance()->find("zichtbaar = 1");
 		echo '<table>';
 		$opties = array(1 => 'nee', 2 => 'misschien', 3 => 'ja');
-		foreach ($voorkeuren as $cid => $voork) {
-			echo '<tr><td>' . $commissies[$cid] . '</td><td>' . $opties[$voork] . '</td></tr>';
+		foreach ($commissies as $commissie) {
+		    $voorkeur = $voorkeurenMap[$commissie->id];
+			echo '<tr><td>' . $commissie->naam . '</td><td>' . $opties[$voorkeur === null ? 1 : $voorkeur->voorkeur] . '</td></tr>';
 		}
 		echo '</table><br />';
-		echo '<h3>Lid opmerkingen</h3><p>' . $voorkeur->getLidOpmerking() . '</p>';
+		echo '<h3>Lid opmerkingen</h3><p>' . $opmerking->lidOpmerking . '</p>';
 		echo '
-		<form name="opties" action="/commissievoorkeuren/lidpagina/' . $this->id . '" method="POST">
-			<textarea name = "opmerkingen" cols=40 rows = 10 >' . $voorkeur->getPraesesOpmerking() . ' </textarea> <br />
+		<form name="opties" action="/commissievoorkeuren/lidpagina/' . $this->profiel->uid . '" method="POST">
```

Comment:
Probeer hier de `Formulier` class te gebruiken. Een `(new CommissieVoorkeurOpmerkingForm())->view()` zou hier al volstaan. Dan kun je door het formulier te validaten checken of je iets moet doen.

En dan kop je gelijk #18 in :joy: